src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -1215,30 +1215,40 @@ private function verifyTypes(string $type, mixed $value, ?array &$invalidTypes =
      */
     private function splitOutsideParenthesis(string $type): array
     {
+        if (!str_contains($type, '|')) {
+            return [$type];
+        }
+
+        if (!str_contains($type, '(') && !str_contains($type, ')')) {
```

Comment:
We could likely optimize that too by looping over characters and check for both parenthesis. This will prevent looping two times.

A regexp to absous the two calls to `str_contains` may also be faster than the current implementation (but slower that the loop I propose).